### PR TITLE
Issue: #503 Allow to serve API without WebUI files

### PR DESCRIPTION
**Pull request by jtorvald**
Added two flags/config options for host_ui and host_docs default to true for backwards compatibility. And fixed link to documentation overview from the UI (pointed to readme, which gives an index of the docs).

### CODE REVIEW

File: internal/http/http.go [hunk 1]
```diff
@@ -152,9 +152,21 @@ func SetupRoutes(handle *graph.Handle, cfg *config.Config) {
...
 	//m.Use(martini.Static("static", martini.StaticOptions{Prefix: "/static", SkipLogging: true}))
 	//r.Handler("GET", "/static", http.StripPrefix("/static", http.FileServer(http.Dir("static/"))))
-	r.GET("/docs/:docpage", docs.ServeHTTP)
-	r.GET("/ui/:ui_type", root.ServeHTTP)
-	r.GET("/", root.ServeHTTP)
-	http.Handle("/static/", http.StripPrefix("/static", http.FileServer(http.Dir(fmt.Sprint(assets, "/static/")))))
-	http.Handle("/", r)
+
+	// only show docs when allowed
+	if cfg.HostDocs {
```
Comment (dennwc) [inline internal/http/http.go]: It's might be better to make it a string flag (path to docs directory). And if it's empty docs will not be server.

Comment (jtorvald) [reply internal/http/http.go]: For the path in the API or the path on disk?
And what about the flag for the UI?

**Comment by dennwc** [comment]
On disk. And the same for UI files, sure.

пт, 17 февр. 2017 г. в 0:41, Jørgen <notifications@github.com>:

File: cmd/cayley/cayley.go [hunk 2]
```diff
@@ -115,15 +115,21 @@ func init() {
...
+	// store
 	rootCmd.PersistentFlags().StringP("db", "d", "memstore", "database backend to use")
 	rootCmd.PersistentFlags().StringP("dbpath", "a", "", "path or address string for database")
 	rootCmd.PersistentFlags().Bool("read_only", false, "open database in read-only mode")
 
-	rootCmd.PersistentFlags().Bool("host_ui", true, "host web ui")
-	rootCmd.PersistentFlags().Bool("host_docs", true, "host documentation")
+	// http
+	rootCmd.PersistentFlags().String("listen", "", "host to listen on")
+	rootCmd.PersistentFlags().String("path_ui", "/ui/", "host web ui")
```
Comment (dennwc) [inline cmd/cayley/cayley.go]: should be `./ui/` and `./docs`

File: cmd/cayley/command/http.go [hunk 3]
```diff
@@ -59,23 +63,43 @@ func NewHttpCmd() *cobra.Command {
...
-			}
 
-			clog.Infof("listening on %s, web interface at http://%s", host, phost)
-			return http.ListenAndServe(host, nil)
+			clog.Infof("listening on %s, web interface at http://%s", split_host, listen)
+			return http.ListenAndServe(listen, nil)
 		},
 	}
-	cmd.Flags().String("host", ":64210", "host:port to listen on")
+	cmd.Flags().String("listen", "", "host:port to listen on")
```
Comment (dennwc) [inline cmd/cayley/command/http.go]: please leave it as `host`, I think it's more common to name it like this.

File: cmd/cayley/command/http.go [hunk 4]
```diff
@@ -59,23 +63,43 @@ func NewHttpCmd() *cobra.Command {
...
+			cmd_host, _ := cmd.Flags().GetString("listen")
+
+			// command overrides configuration file
+			if cmd_host != "" {
+				listen = cmd_host
+			}
+
+			// do we have a port, otherwise add default port
+			var findPort = regexp.MustCompile(`:[0-9]+$`)
+			if !findPort.MatchString(listen) {
```
Comment (dennwc) [inline cmd/cayley/command/http.go]: SplitHostPort should work fine here

File: internal/http/http.go [hunk 5]
```diff
@@ -145,27 +145,27 @@ func SetupRoutes(handle *graph.Handle, cfg *config.Config) {
 	}
 	var templates = template.Must(template.ParseGlob(fmt.Sprint(assets, "/templates/*.tmpl")))
 	templates.ParseGlob(fmt.Sprint(assets, "/templates/*.html"))
-	root := &TemplateRequestHandler{templates: templates}
-	docs := &DocRequestHandler{assets: assets}
+	root := &TemplateRequestHandler{templates: templates, path: cfg.HostUI}
+	docs := &DocRequestHandler{assets: assets, path: cfg.HostDocs}
```
Comment (dennwc) [inline internal/http/http.go]: Assets is the same parameter, no?

**Pull request closed without merging**